third_party/move/move-binary-format/src/deserializer.rs

```diff
@@ -1810,3 +1906,66 @@ impl Opcodes {
         }
     }
 }
+
+impl SerializedBool {
+    fn from_u8(value: u8) -> BinaryLoaderResult<bool> {
+        match value {
+            0x1 => Ok(false),
+            0x2 => Ok(true),
+            _ => Err(PartialVMError::new(StatusCode::MALFORMED)
+                .with_message("malformed boolean".to_owned())),
+        }
+    }
+}
+
+impl SerializedOption {
+    fn from_u8(value: u8) -> BinaryLoaderResult<bool> {
```

Comment:
This isn't great.  What do `true` and `false` mean as `Option` enum discriminants?  Better if you can have an enum somewhere for the values.

I found this related crate: https://docs.rs/enum-kinds/latest/enum_kinds/ but you can't use it for Option, anyway, so it wouldn't be useful even if you wanted to use it.  `std::mem::Discriminant<Option<X>>` might make sense here, but without `X` I don't think it's meaningful.  But using 'bool' is too confusing.

Reply:
This is a boolean indicating the presence or the absence of the option. I think is fine.